app/src/main/java/org/eyeseetea/malariacare/domain/entity/Question.java

```diff
@@ -17,41 +19,24 @@
     private Visibility visibility;
     private List<Rule> rules;
 
-    public Question() {
-    }
-
     public Question(long id, String code, String name, String uid,
             PhoneFormat phoneFormat, Type type, boolean compulsory,
             List<Option> options, Header header, int index,
             Visibility visibility,
             List<Rule> rules) {
-        this.id = id;
-        this.code = code;
-        this.name = name;
-        this.uid = uid;
-        this.phoneFormat = phoneFormat;
-        this.type = type;
-        this.compulsory = compulsory;
-        this.options = options;
-        this.header = header;
-        this.index = index;
-        this.visibility = visibility;
-        this.rules = rules;
-    }
 
-    private Question(Builder builder) {
-        setId(builder.id);
-        setCode(builder.code);
-        setName(builder.name);
-        setUid(builder.uid);
-        setPhoneFormat(builder.phoneFormat);
-        setType(builder.type);
-        setCompulsory(builder.compulsory);
-        setOptions(builder.options);
-        setHeader(builder.header);
-        setIndex(builder.index);
-        setVisibility(builder.visibility);
-        setRules(builder.rules);
+        setId(required(id, "id is required"));
```

Comment:
Can you remove private setters? It's not necessary we have private setters, so we reduce the size of the class